### PR TITLE
pip package for the pyang plugin

### PR DESCRIPTION
**Pull request by martin-volf**
Changes necessary so that we can start pushing the jnc package to PyPI. The change also affect a bit how examples and tests are built (note they use the plugin from the source, not from PyPI).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (jomiklos) [inline setup.cfg:1]: redundant empty leading line?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+[metadata]
+name = jnc
+summary = Pyang plugin generating classes for Java Netconf Client
```
Comment (jomiklos) [inline setup.cfg:4]: maybe we should stabilize JNC name with capital NETCONF lettering across repo, not to introduce "incorrect" Netconf form?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+
+from setuptools import setup
```
Comment (jomiklos) [inline setup.py:3]: doesn't `requirements.txt` need `setuptools`?

**Review by jojohans — APPROVED** [review]

**Review by micnovak — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (martin-volf) [reply setup.cfg]: fixed

File: setup.cfg (hunk 2 above)
Comment (martin-volf) [reply setup.cfg]: fixed

File: setup.py (hunk 3 above)
Comment (martin-volf) [reply setup.py:3]: No, this is needed only for building the distribution package, not for using the plugin.

File: tests/1-test-union/build.gradle [hunk 4]
```diff
@@ -5,7 +5,7 @@ plugins {
 assert (System.env.CONFD_DIR): "CONFD_DIR is not set!"
 
 task pyang(type: JncPyangTask) {
-    pluginDir = JNCDir
+    pluginDir = PyangJNC
```
[the remark this answers is not in the document]
Comment (martin-volf) [reply tests/1-test-union/build.gradle:8]: Yes, once we publish on PyPI, this and possibly few more changes will be needed.

**Review by jomiklos — APPROVED** [review]

**Pull request merged**